File: westcoast/src/main/java/frc/commands/BallHandlers/IntakeAndIndex.java
package frc.commands.BallHandlers;

import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
import frc.subsystems.BallHandlingSubsystem;

public class IntakeAndIndex extends SequentialCommandGroup {
  private final BallHandlingSubsystem ball;

  public IntakeAndIndex(BallHandlingSubsystem ball) {
    addCommands(
      new IntakeLogic(ball),
      new IndexLogic(ball)
    );
    this.ball = ball;
    addRequirements(ball);
  }

  @Override
  public void end(boolean interrupted) {
    System.out.println("Ball Implemented!");
    ball.stopMotors();
  }
}

File: westcoast/src/main/java/frc/commands/ballhandling/IntakeAndIndex.kt
package frc.commands.ballhandling

import edu.wpi.first.wpilibj2.command.SequentialCommandGroup
import edu.wpi.first.wpilibj2.command.WaitCommand
import frc.subsystems.BallHandlingSubsystem

class IntakeAndIndex(ball: BallHandlingSubsystem) : SequentialCommandGroup() {
  private val ball: BallHandlingSubsystem
  override fun end(interrupted: Boolean) {
    println("Ball Implemented!")
    ball.stopMotors()
  }

  init {
    addCommands(
        IntakeLogic(ball),
        WaitCommand(0.2),
        IndexLogic(ball)
    )
    this.ball = ball
    addRequirements(ball)
  }
}